src/engine/plugins/tile.cpp

```diff
@@ -380,226 +387,218 @@ Status TilePlugin::HandleRequest(const std::shared_ptr<datafacade::BaseDataFacad
         // are all pointed in the same direction.
         struct EdgeBasedNodeInfo
         {
-            unsigned source_intersection; // node-based-node ID
-            unsigned target_intersection; // node-based-node ID
+            bool is_geometry_forward; // Is the geometry forward or reverse?
             unsigned packed_geometry_id;
         };
         // Lookup table for edge-based-nodes
         std::unordered_map<NodeID, EdgeBasedNodeInfo> edge_based_node_info;
-        std::unordered_map<NodeID, std::vector<EdgeID>> outgoing_edges;
-        std::unordered_map<NodeID, std::vector<EdgeID>> incoming_edges;
 
-        // Now, loop over all the road segments we saw, and build up a mini
-        // graph for just the network that's visible.
+        struct SegmentData
+        {
+            NodeID target_node;
+            EdgeID edge_based_node_id;
+        };
+
+        std::unordered_map<NodeID, std::vector<SegmentData>> directed_graph;
+
+        // Build an adjacency list for all the road segments visible in
+        // the tile
         for (const auto &edge : edges)
         {
-            // Note: edge.u is the node-based node ID of the source intersection
-            //       edge.v is the node-based node ID of the target intersection
-            //       both these values can be directly looked up for coordinates
-
-            // If forward travel is enabled on this road section, and we haven't seen this
-            // edge-based-node
-            // before
-            if (edge.forward_segment_id.enabled &&
-                edge_based_node_info.count(edge.forward_segment_id.id) == 0)
+            if (edge.forward_segment_id.enabled)
             {
-                // Add this edge-based-nodeid as an outgoing from the source intersection
-                const auto outgoing_itr_u = outgoing_edges.find(edge.u);
-                if (outgoing_itr_u != outgoing_edges.end())
-                {
-                    outgoing_itr_u->second.push_back(edge.forward_segment_id.id);
-                }
-                else
+                if (directed_graph.count(edge.u) == 0)
                 {
-                    outgoing_edges[edge.u] = {edge.forward_segment_id.id};
-                }
-
-                // Add this edge-based-nodeid as an incoming to the target intersection
-                const auto incoming_itr_v = incoming_edges.find(edge.v);
-                if (incoming_itr_v != incoming_edges.end())
-                {
-                    incoming_itr_v->second.push_back(edge.forward_segment_id.id);
+                    directed_graph[edge.u] = {{edge.v, edge.forward_segment_id.id}};
```

Comment:
is this distinction `if/else` necessary? The `[]` operator should create the initial vector on access.